### ISSUE TITLE
Add MCTS as an option for the AI

### ISSUE THREAD

**Issue by allentiak** [magefree/mage#7075]
Considering #6114, I think it would be nice to add MCTS AI as an option.

For inspiration, a screenshot of similar functionality in Magarena.

Magarena offers Las Vegas, MiniMax and MCTS (in all cases, both with and without cheating).

![Magarena 1.96 -- Select AI](https://user-images.githubusercontent.com/1922297/93785652-b67dfc00-fc04-11ea-8f41-cfb7b1f29f5a.png)

**Comment by jeffwadsworth** [comment]
It is there, but incredibly slow and buggy.

**Comment by allentiak** [comment]
By "as an option" I meant "to the UI". This would have the added benefit of
enabling casual testing of the AI.

**Comment by JayDi85** [comment]
It must be tested by someone -- is it playable or not in real games (I don't want to enable it in public or beta servers). If MCTS AI can complete 50% of all games then it can be enabled by default in local configs.

Also it can be useful to add AI game stats (total AI games, win/lose).

**Comment by allentiak** [comment]
> If MCTS AI can complete 50% of all games then it can be enabled by default in local configs.

I agree: if MCTS is enabled as an option for local servers (along with
the existing one, of course), it will be easier to improve it.

More players ==> More testers

**Comment by SpeedProg** [comment]
I activated it on my local server to play against it, and it was throwing exceptions all the time. I barely ever got through a step without an except getting thrown. I fixed the exception in a none proper way, and the AI felt way worse then the normal minmax or what it is called (it did so too when I spent hours rolling back on exceptions on the none "fixed" one). There is really no reason to enable it in its current state, except to annoy people that use it by accident. Some one would need to completely overhaul it I am pretty sure of that.

**Comment by allentiak** [comment]
Maybe it could be activated with an ALPHA warning? My point is to get people to use it; hopefuly, that could bring volunteers to work on it.

BTW: The only libre AI MCTS implementation I am aware of, is Magarena's. _(Maybe it could be integrated into XMage, but that would require XMage to switch to the GPL.)_  [[License-related discussion]](https://github.com/magefree/mage/issues/7010#issuecomment-689007008)
